### PR TITLE
docs: extend docs with the different node configuration mechanisms

### PR DESCRIPTION
**Pull request by LNSD**
I am not 100% sure if this is the right place (the correct markdown file) to provide these details. If you know a better place where to specify this information, do not hesitate to suggest it in the comments.

- [x] Extend operator's documentation with the different configuration methods supported by `wakunode2`

### CODE REVIEW

**Comment by LNSD** [comment]
@mfw78, it would be interesting to hear your opinion on this and if this covers your requirements hence #1198

File: docs/operators/how-to/configure.md [hunk 1]
```diff
@@ -1,12 +1,135 @@
...
+## Node configuration methods
+
+One node can be configured using a combination of the following methods:
+
+1. Command line options and flags
+2. Environment variables
+3. Configuration file (currently, only TOML format is supported)
+4. Default value
+
+Note the precedence order, each item takes precedence over the item below it.
```
Comment (alrevuelta) [inline docs/operators/how-to/configure.md:16]: I had to read this couple of times `precedence over the item below it`. Perhaps be more explicit? "each item overrides the one below."

**Review by alrevuelta — APPROVED** [review]
lgtm! thanks for the PR! really relevant. left a minor comment.

File: docs/operators/how-to/configure.md (hunk 1 above)
Comment (LNSD) [reply docs/operators/how-to/configure.md:16]: How would you rephrase it? What do you suggest?

Comment (alrevuelta) [reply docs/operators/how-to/configure.md:16]: > "each item overrides the one below."

Note the precedence order, where "each item overrides the one below."

**Review by mfw78 — APPROVED** [review]
LGTM. Great work on the configuration options! This fits perfectly with the issue that was raised 👍

File: docs/operators/how-to/configure.md [hunk 2]
```diff
@@ -1,12 +1,135 @@
...
+```toml
+log-level = "DEBUG"
+tcp-port = 65000
+```
+
+The path to the TOML file can be specified usin one of the previous configuration mechanisms:
+
+* By passing the `--config-file` command line option:
+  ```shell
+  waknode2 --config-file=<path-to-toml-config-file>
```
Comment (rymnc) [inline docs/operators/how-to/configure.md:97]:
```suggestion
  wakunode2 --config-file=<path-to-toml-config-file>
```

**Review by rymnc — APPROVED** [review]
LGTM, this looks like a good place to put it, maybe can go in its own how-to later

**Review by alrevuelta — APPROVED** [review]
lgtm! thanks for the example on the overrides: i.e. cli overrides env.

**Pull request merged**